server/src/main/java/org/candlepin/policy/js/entitlement/AbstractEntitlementRules.java

```diff
@@ -270,19 +288,33 @@ public String toString() {
 
     protected void runPostEntitlement(PoolManager poolManager, Consumer consumer,
         Map<String, Entitlement> entitlementMap, List<Pool> subPoolsForStackIds) {
+
         Map<String, Map<String, String>> flatAttributeMaps = new HashMap<String, Map<String, String>>();
         Map<String, Entitlement> virtLimitEntitlements = new HashMap<String, Entitlement>();
+        Map<String, Entitlement> sharedEntitlements = new HashMap<String, Entitlement>();
+
         for (Entry<String, Entitlement> entry : entitlementMap.entrySet()) {
             Entitlement entitlement = entry.getValue();
             Map<String, String> attributes = PoolHelper.getFlattenedAttributes(entitlement.getPool());
             if (attributes.containsKey("virt_limit")) {
                 virtLimitEntitlements.put(entry.getKey(), entitlement);
                 flatAttributeMaps.put(entry.getKey(), attributes);
             }
+            if (consumer.getType().isType(ConsumerType.ConsumerTypeEnum.SHARE)) {
+                sharedEntitlements.put(entry.getKey(), entitlement);
+                flatAttributeMaps.put(entry.getKey(), attributes);
+            }
         }
         // Perform pool management based on the attributes of the pool:
-        postBindVirtLimit(poolManager, consumer, virtLimitEntitlements, flatAttributeMaps,
-            subPoolsForStackIds);
+        if (!virtLimitEntitlements.isEmpty()) {
```

Comment:
@vritant I've addressed this in the lines below the one you commented on.  I put the logic in a separate `if` because combining all the predicates together would, in my opinion, be very difficult to decipher.